src/components/faktum/FaktumDato.test.tsx

```diff
@@ -63,14 +61,244 @@ describe("FaktumDato", () => {
-        expect(onchange).toBeCalledWith(faktumMockData, svar);
+        expect(onchange).toBeCalledWith(faktumMockData, null);
+      });
+    });
+  });
+
+  describe("When user selects a date three weeks from now", () => {
+    test("Should post selected date to server and display warning message", async () => {
+      const threeWeeksFromNow = addWeeks(new Date(), 3);
+      const datePickerFormattedDate = format(threeWeeksFromNow, "dd.MM.yyyy"); // eg: 20.11.2022
+      const isoFormattedDate = formatISO(threeWeeksFromNow, { representation: "date" }); // eg 2022-11-20
+
+      const user = userEvent.setup();
+      const onchange = jest.fn();
+
+      render(
+        <MockContext>
+          <FaktumDato faktum={faktumMockData} onChange={onchange} />
+        </MockContext>
+      );
+
+      const datepicker = screen.getByLabelText(faktumMockData.beskrivendeId) as HTMLInputElement;
+      await user.type(datepicker, datePickerFormattedDate);
+      const warningMessage = screen.getByTestId("faktum.soknadsdato-varsel");
+
+      await waitFor(() => {
+        expect(onchange).toBeCalledTimes(1);
+        expect(onchange).toBeCalledWith(faktumMockData, isoFormattedDate);
+        expect(warningMessage).toBeInTheDocument();
+      });
+    });
+  });
+
+  describe("When user selects a date over 100 years from now", () => {
+    test("Should show error message", async () => {
+      const twoHundretYearsFromNow = addYears(new Date(), 200);
+      const datePickerFormattedDate = format(twoHundretYearsFromNow, "dd.MM.yyyy");
+
+      const user = userEvent.setup();
+      const onchange = jest.fn();
+
+      render(
+        <MockContext>
+          <FaktumDato faktum={faktumMockData} onChange={onchange} />
+        </MockContext>
+      );
+
+      const datepicker = screen.getByLabelText(faktumMockData.beskrivendeId) as HTMLInputElement;
+      await user.type(datepicker, datePickerFormattedDate);
+      const datePickerError = document.querySelector(
+        ".navds-form-field__error"
+      ) as HTMLInputElement;
+      await user.type(datepicker, datePickerFormattedDate);
+
+      await waitFor(() => {
+        expect(datePickerError).toBeInTheDocument();
+      });
+    });
+  });
+
+  describe("When user types in 10102022 on datepicker", () => {
+    test("Should post 2022-10-10 to server", async () => {
+      const user = userEvent.setup();
+      const onchange = jest.fn();
+
+      render(
+        <MockContext>
+          <FaktumDato faktum={faktumMockData} onChange={onchange} />
+        </MockContext>
+      );
+
+      const datepicker = screen.getByLabelText(faktumMockData.beskrivendeId) as HTMLInputElement;
+      await user.type(datepicker, "10102022");
+
+      await waitFor(() => {
+        expect(onchange).toBeCalledTimes(1);
+        expect(onchange).toBeCalledWith(faktumMockData, "2022-10-10");
+      });
+    });
+  });
+
+  describe("When user removes a date three weeks from now", () => {
```

Comment:
Legge denne i samme describe som resten av testene for faktum.dagpenger-soknadsdato? Describe kan sees på som en kategorisering av tester som hører sammen. :)